collector/cfservicescollector.go

```diff
@@ -38,14 +38,66 @@ func (c *CfServicesCollector) GetAnnotations() []string {
 	return annotations
 }
 
+// getNewIbmcloudCfClient returns a new ibmcloud cf client
+func (c *CfServicesCollector) getNewIbmcloudCfClient(homeDir string) (*cfclient.Client, error) {
+	var client *cfclient.Client
+	cfclientConfig, err := cfclient.NewConfigFromCFHome(filepath.Join(homeDir, ".ibmcloud/"))
+	if err == nil {
+		client, err = cfclient.NewClient(cfclientConfig)
+		if err != nil {
+			logrus.Debugf("The .ibmcloud directory based cf login failed while creating new client. Falling back to the .bluemix directory based login")
+			cfclientConfig, err = cfclient.NewConfigFromCFHome(filepath.Join(homeDir, ".bluemix/"))
+			if err == nil {
+				client, err = cfclient.NewClient(cfclientConfig)
+				if err != nil {
+					logrus.Debugf("The .bluemix directory based cf login failed while creating new client.")
+				}
+			} else {
+				logrus.Debugf("Unable to get cf config from .bluemix directory : %s", err)
+			}
+		}
+	} else {
+		logrus.Debugf("Unable to get cf config from .ibmcloud directory : %s", err)
+		logrus.Debugf("The .ibmcloud directory based cf login failed. Falling back to the .bluemix directory based login")
+		cfclientConfig, err = cfclient.NewConfigFromCFHome(filepath.Join(homeDir, ".bluemix/"))
+		if err == nil {
+			client, err = cfclient.NewClient(cfclientConfig)
+			if err != nil {
+				logrus.Debugf("The .bluemix directory based cf login failed while creating new client.")
+			}
+		} else {
+			logrus.Debugf("Unable to get cf config from .bluemix directory : %s", err)
+		}
+	}
+	return client, err
+}
+
+// getNewClient returns a new client
+func (c *CfServicesCollector) getNewClient(homeDir string) (*cfclient.Client, error) {
```

Comment:
Probably create a file named say cfutil and move this function to that file, so that we need not repeat the code twice

Reply:
Done